### PR TITLE
Define new contract for uploading validation results & request together

### PR DESCRIPTION
**Pull request by rpatel-figure**
## Context
The current `RecordLoanValidationResultsContract` is designed for interactions between two parties where one is submitting requests for validation and the other is submitting their fulfillment of those requests. However, for cases where a single party wishes to submit validation results, the contract is not suitable. For this usecase, it would be ideal to define a new contract where the submission is not required to match against an existing request in the record, and simply adds another validation iteration which contains both a request and results.
## Changes
- Define a new contract for uploading a combined validation request and results
- Add missing test cases to enforce the distinction between the now two different contracts for uploading validation results
- Remove requirement for validation requests to specify block height
- Fix failing funding contract unit test edge case

### CODE REVIEW

File: contract/src/main/kotlin/io/provenance/scope/loan/contracts/RecordStandaloneLoanValidationResultsContract.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import io.provenance.scope.loan.utility.ContractRequirementType
+import io.provenance.scope.loan.utility.loanValidationResultsValidation
+import io.provenance.scope.loan.utility.validateRequirements
+import tech.figure.validation.v1beta2.LoanValidation
+import tech.figure.validation.v1beta2.ValidationIteration
+import tech.figure.validation.v1beta2.ValidationResponse
+
+@Participants(roles = [PartyType.VALIDATOR])
+@ScopeSpecification(["tech.figure.loan"])
+open class RecordStandaloneLoanValidationResultsContract(
```
Comment (dkneisly-figure) [inline contract/src/main/kotlin/io/provenance/scope/loan/contracts/RecordStandaloneLoanValidationResultsContract.kt:21]: 🤣 names are hard

File: contract/src/main/kotlin/io/provenance/scope/loan/contracts/RecordStandaloneLoanValidationResultsContract.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        @Input(LoanScopeInputs.validationResponse) submission: ValidationResponse
+    ): LoanValidation {
+        validateRequirements(ContractRequirementType.VALID_INPUT) {
+            loanValidationResultsValidation(submission.results)
+        }
+        return (validationRecord?.toBuilder() ?: LoanValidation.newBuilder()).also { recordBuilder ->
+            recordBuilder.addIteration(
+                ValidationIteration.newBuilder().also { iterationBuilder ->
+                    iterationBuilder.results = submission.results
+                }.build()
```
Comment (dkneisly-figure) [inline contract/src/main/kotlin/io/provenance/scope/loan/contracts/RecordStandaloneLoanValidationResultsContract.kt:37]: Food for thought: do you think we should make the invoker pass a ValidationRequest as well? That way we could at least fill in a description, requester/validator to give more context to people auditing this data later.

**Review by dkneisly-figure — COMMENTED** [review]
Overall looks great, just want to get your thoughts on the concept of requiring both the request and response in this new contract for auditing purposes.

File: contract/src/main/kotlin/io/provenance/scope/loan/contracts/RecordStandaloneLoanValidationResultsContract.kt (hunk 2 above)
Comment (rpatel-figure) [reply contract/src/main/kotlin/io/provenance/scope/loan/contracts/RecordStandaloneLoanValidationResultsContract.kt:37]: I suppose so. Would we then also want to add a field / set a value somewhere to indicate that the request & response were set by the same party?

**Review by dkneisly-figure — APPROVED** [review]
🚀
